pkg/services/ngalert/api/api_contact_points.go

```diff
@@ -0,0 +1,71 @@
+package api
+
+import (
+	"errors"
+	"net/http"
+
+	"github.com/grafana/grafana/pkg/api/response"
+	api "github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/services/ngalert/services"
+	"github.com/grafana/grafana/pkg/web"
+)
+
+type ContactPointServer struct {
+	service services.ContactPointService
+}
+
+func (s *ContactPointServer) RouteGetContactPoint(c *api.ReqContext) response.Response {
+	if !c.HasUserRole(api.ROLE_VIEWER) {
+		return ErrResp(http.StatusForbidden, errors.New("permission denied"), "")
+	}
+	templates, err := s.service.GetContactPoints(c.OrgId)
+	if err != nil {
+		return ErrResp(http.StatusInternalServerError, err, "")
+	}
+	return response.JSON(http.StatusOK, templates)
+}
+
+func (s *ContactPointServer) RouteCreateContactPoint(c *api.ReqContext) response.Response {
+	if !c.HasUserRole(api.ROLE_EDITOR) {
+		return ErrResp(http.StatusForbidden, errors.New("permission denied"), "")
+	}
+	contactPoint := services.EmbeddedContactPoint{}
+	if err := web.Bind(c.Req, &contactPoint); err != nil {
+		return response.Error(http.StatusBadRequest, "bad request data", err)
+	}
+	contactPoint, err := s.service.CreateContactPoint(c.OrgId, contactPoint)
+	if err != nil {
+		return ErrResp(http.StatusInternalServerError, err, "")
+	}
+	return response.JSON(http.StatusOK, contactPoint)
+}
+
+func (s *ContactPointServer) RouteUpdateContactPoint(c *api.ReqContext) response.Response {
+	if !c.HasUserRole(api.ROLE_EDITOR) {
+		return ErrResp(http.StatusForbidden, errors.New("permission denied"), "")
+	}
+	contactPoint := services.EmbeddedContactPoint{}
+	if err := web.Bind(c.Req, &contactPoint); err != nil {
+		return response.Error(http.StatusBadRequest, "bad request data", err)
+	}
+	err := s.service.UpdateContactPoint(c.OrgId, contactPoint)
+	if err != nil {
+		return ErrResp(http.StatusInternalServerError, err, "")
+	}
+	return response.JSON(http.StatusOK, "")
+}
+
+func (s *ContactPointServer) RouteDeleteContactPoint(c *api.ReqContext) response.Response {
+	if !c.HasUserRole(api.ROLE_EDITOR) {
+		return ErrResp(http.StatusForbidden, errors.New("permission denied"), "")
+	}
+	contactPoint := services.EmbeddedContactPoint{}
+	if err := web.Bind(c.Req, &contactPoint); err != nil {
```

Comment:
TODO: It's unnatural to have a DELETE route accept body info. RESTfully, "DELETE /collection" implies deleting the whole collection (i.e. all contact points) regardless of the body. Should probably route `DELETE /api/alerting/contact-points/<uid>` to this method instead and just read the one field from the URL